src/wallet/mod.rs

```diff
@@ -393,6 +393,14 @@ where
         };
 
         let mut fee_amount = fee_amount.ceil() as u64;
+
+        if builder.has_absolute_fee() {
+            fee_amount = match builder.fee_policy.as_ref().unwrap() {
+                FeePolicy::FeeAmount(amount) => *amount,
+                _ => fee_amount,
+            }
+        };
```

Comment:
See comment above, I think this should be moved up before the coin selection.

I would actually do the `fee_rate`/`fee_amount` initialization in one go. Something like:

```rust
let (fee_rate, mut fee_amount) = match builder.fee_policy.as_ref().unwrap_or(FeePolicy::FeeRate(FeeRate::default())) {
    FeePolicy::FeeAmount(amount) => (FeeRate::from_sat_per_vb(0.0), amount as f32),
    FeePolicy::FeeRate(rate) => (rate, 0.0),
};
```

In my opinion, it also makes sense to implement `Default` for `FeePolicy`, which would return `FeePolicy::FeeRate(FeeRate::default())`. With that you could then turn the `.unwrap(...)` above to just `.unwrap_or_default()`.

As a bonus, this would also remove the need for the `get_fee_rate()` function.